pymeasure/instruments/hp/hp437b.py

```diff
@@ -0,0 +1,436 @@
+        power meter to report the pulse power of a rectangular pulsed input signal. The
+        allowable range of values for 'duty_cycle' is 0.001 to 99.999%.
+
+        Pulse power, as reported by the power meter, is a mathematical
+        representation of the pulse power rather than an actual measurement.
+        The power meter measures the average power of the pulsed input
+        signal and then divides the measurement by the duty cycle value to
+        obtain a pulse power reading.
+        """,
+        validator=lambda v, vs: strict_discrete_range(v, vs, 0.001),
+        values=[0.001, 99.999],
+        check_set_errors=True
+    )
+
+    # status_message = Instrument.measurement(
+    #     "SM", "",
+    #     """
+    #     """
+    # )
+
+    filter_automatic_enabled = Instrument.control(
+        "SM", "%s",
+        """
+        Control the filter mode. By switching over from automatic to manual (true to false)
+        the instrument implicitly keeps (holds) the filter value from the automatic selection.
+        """,
+        cast=bool,
+        get_process=__getstatus(StatusMessage.AutoFilterStatus),
+        set_process=lambda v: "FA" if v else "FH",
+        check_set_errors=True
+    )
+
+    filter = Instrument.control(
+        "SM", "FM%dEN",
+        """
+        Control the filter number for averaging. Setting a value implicitly enables the manual 
+        filter mode.
+        """,
+        values=[1, 2, 4, 8, 16, 32, 64, 128, 256, 512],
+        validator=strict_discrete_set,
+        get_process=__getstatus(StatusMessage.Filter, (lambda x: 2 ** x)),
+        check_set_errors=True
+    )
+
+    frequency = Instrument.setting(
+        "FR%08.4fGZ",
+        """
+        Setthe frequency of the input signal. Entering a frequency causes the power meter to select
+        a sensor-specific calibration factor. The allowable range of 'frequency'
+        values is from 0.0001 to 999.9999 GHz with a 100 kHz resolution.
+        """,
+        set_process=lambda v: v / 1e9,
+        check_set_errors=True
+    )
+
+    limits_enabled = Instrument.control(
+        "SM", "LM%d",
+        """
+        Control the limits checking function to allow the power meter to monitor the
+        power level at the sensor and to indicate when that power is outside
+        preset limits.
+        """,
+        map_values=True,
+        values={True: 1, False: 0},
+        cast=int,
+        get_process=__getstatus(StatusMessage.LimitsCheckingStatus),
+        check_set_errors=True
+    )
+
+    limit_high = Instrument.setting(
+        "LH%5.2fEN",
+        """
+        Set the upper limit for the builtin limit checking.
+        """,
+        check_set_errors=True
+    )
+
+    limit_low = Instrument.setting(
+        "LL%5.2fEN",
+        """
+        Set the lower limit for the builtin limit checking.
+        """,
+        check_set_errors=True
+    )
```

Comment:
Two comments on these limit functions;
1. without reading the documentation it might be difficult for the user to understand the value range which could be applied.
2. would it make sense to have validators in these functions?